### DISCUSSION TITLE
How to set WPF WebView2 FlowDirection correctly?

### DISCUSSION THREAD

**Discussion by Tennyleaz** [MicrosoftEdge/WebView2Feedback#3874] [Q&A]
I would like to set webview2 FlowDirection from right to left at some scenarios:

```C#
string localeName = ...
switch (localeName)
{
    default:
        myWebView2.FlowDirection = FlowDirection.LeftToRight;
        break;
    case "ar":
    case "fa":
    case "ps":
    case "ur":
        myWebView2.FlowDirection = FlowDirection.RightToLeft;
        break;
}
```
But the html display on my webview2 still always start from the left.
How could I set FlowDirection correctly?

**Comment by champnic** [comment]
Hey @Tennyleaz - this is an interesting question, and I'm not very familiar with what browser do by default today in RTL scenarios, and whether or not they automatically adapt. If you show your web content in Edge or Chrome and use an RTL language, do you see the html automatically changing flow direction? If not, it's possible you will need to edit your web content (if you own the web content) to use the 'dir' attribute (https://html.spec.whatwg.org/#the-dir-attribute) or potentially ExecuteScript on the page that tells it to change direction.

**Comment by Tennyleaz** [comment]
Hi champnic, the html is mostly user-generated document (not by me), which may contain RTL language scripts. 
If using Chrome or Edge, the html file does not change direction automatically.
It's is difficult to inject `dir=rtl` tag properly into the html file, so I am seeking a more general solution here.

**Comment by champnic** [comment]
@Tennyleaz Are you changing the language/locale while Chrome/Edge/WebView2 is running, or is it still broken when launching it? You could try setting the language when creating the environment for the WebView2 (note: you wouldn't be able to use "Source" to initialize the WebView2 - instead you would create the CoreWebView2EnvironmentOptions with language, then create the CoreWeBView2Environment with those options, then EnsureCoreWebView2Async with the environment)
